infrastructure/controller/pkg/endpoints/enable_disable_components.go

```diff
@@ -0,0 +1,109 @@
+package endpoints
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+)
+
+type EnableDisableComponents struct {
+	clientSet *kubernetes.Clientset
+	namespace string
+}
+
+type component struct {
+	Name   string `json:"name"`
+	Enable bool   `json:"enable,string"`
+}
+
+func (s *EnableDisableComponents) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	blob, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		klog.Infof("not able to read request body %s", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	var c component
+	if err := json.Unmarshal(blob, &c); err != nil {
+		klog.Errorf("invalid payload %s", err)
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	deployments, err := s.clientSet.AppsV1().Deployments(s.namespace).List(
+		r.Context(),
+		v1.ListOptions{LabelSelector: fmt.Sprintf("core.airy.co/component=%s", c.Name)},
+	)
+
+	for _, deployment := range deployments.Items {
+		if c.Enable && *deployment.Spec.Replicas > 0 {
+			w.WriteHeader(http.StatusOK)
+			return
+		}
+
+		if !c.Enable && *deployment.Spec.Replicas == 0 {
+			w.WriteHeader(http.StatusOK)
+			return
+		}
+
+		if c.Enable {
+			*deployment.Spec.Replicas = 1
+		} else {
+			*deployment.Spec.Replicas = 0
+		}
+		//TODO: Use Patch insdead
+		_, err := s.clientSet.AppsV1().Deployments(s.namespace).Update(
```

Comment:
Why are we updating the deployment from the endpoint? Maybe it will be a better approach if the endpoint only changes the configMap/component and then the controller, when he detects this change, reacts and applies the change.